nextcloud/server
# [3rdparty] Doctrine: Fix unquoted stmt fragments backslash escaping

Needed for https://github.com/nextcloud/server/pull/22118 - already in master because there doctrine/dbal:2.10.2 is used


Then #22118 properly works.

* [x] requires https://github.com/nextcloud/3rdparty/pull/486
* brings in https://github.com/doctrine/dbal/pull/3772


How to test:

```
php occ group:add test
php occ user:add test1 --display-name abc --group test
curl 'http://SERVER/ocs/v2.php/cloud/groups/test/users/details?offset=0&limit=25&search=abc' -u admin:password -H "OCS-APIRequest: true"
```

The curl should return the proper result. Before it returned an empty set:

Before:

```xml
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message>OK</message>
 </meta>
 <data>
  <users/>
 </data>
</ocs>
```

After:

```xml
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message>OK</message>
 </meta>
 <data>
  <users>
   <test1>
    <enabled>1</enabled>
    <storageLocation>/Users/morris/Projects/nextcloud/server/data/test1</storageLocation>
    <id>test1</id>
    <lastLogin>0</lastLogin>
    <backend>Database</backend>
    <subadmin/>
    <quota>
     <quota>none</quota>
     <used>0</used>
    </quota>
    <email/>
    <displayname>abc</displayname>
    <phone></phone>
    <address></address>
    <website></website>
    <twitter></twitter>
    <groups>
     <element>test</element>
    </groups>
    <language>en</language>
    <locale></locale>
    <backendCapabilities>
     <setDisplayName>1</setDisplayName>
     <setPassword>1</setPassword>
    </backendCapabilities>
   </test1>
  </users>
 </data>
</ocs>
```
